src/App.tsx

```diff
@@ -1,24 +1,73 @@
-import React from 'react';
+import { Component } from 'react';
+import { Clock } from './Clock';
 import './App.scss';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean,
+  clockName: string,
 };
 
+class App extends Component<{}, State> {
+  state = {
+    isClockVisible: false,
+    clockName: 'Name 1',
+  };
+
+  hideClock = () => {
+    this.setState({
+      isClockVisible: false,
+    });
+  };
+
+  showClock = () => {
+    this.setState({
+      isClockVisible: true,
+    });
+  };
+
+  setRandomName = () => {
+    this.setState({
+      clockName: `Name ${Math.round(Math.random() * 1000)}`,
+    });
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">React clock</h1>
+        <div className="App__clock">
+          {isClockVisible ? <Clock name={clockName} /> : <p>00:00:00</p>}
```

Comment:
Maybe better show text 'Oops' or something like this instead '00:00:00'. What do you think?